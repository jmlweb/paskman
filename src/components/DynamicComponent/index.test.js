import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import createDynamicComponent from './';

configure({ adapter: new Adapter() });

Object.defineProperty(window.navigator, "userAgent", (function(_value){
  return {
    get: function _get() {
      return _value;
    },
    set: function _set(v) {
        _value = v;
    }
  };
})(window.navigator.userAgent));
Object.defineProperty(window, "reactSnapshotRender", (function(_value){
  return {
    get: function _get() {
      return _value;
    },
    set: function _set(v) {
        _value = v;
    }
  };
})(window.reactSnapshotRender));

describe('DynamicComponent', () => {
  let wrapper;
  beforeEach(() => {
    window.navigator.userAgent = 'Node.js';
    window.reactSnapshotRender = true;
    const component = createDynamicComponent(
      () => import('../../scenes/Home'), () => require('../../scenes/Home')
    );
    wrapper = mount(<component />);
  });
  it('should render the dynamic component with require', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });
  it('Matchs snapshot', () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
