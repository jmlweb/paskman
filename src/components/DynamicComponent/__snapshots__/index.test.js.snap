// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicComponent Matchs snapshot 1`] = `<component />`;
